examples/widgets/mainwindow.cpp

```diff
@@ -7,24 +7,25 @@
 #include "window.hpp"
 
 #include <QMenu>
-#include <QMenuBar>
 #include <QMessageBox>
 
 MainWindow::MainWindow() {
-    auto *menuBar = new QMenuBar(this);
-    QMenu *menuWindow = menuBar->addMenu(tr("&Window"));
-    auto *actionAddNew = new QAction(menuWindow);
-    actionAddNew->setText(tr("Add new"));
-    menuWindow->addAction(actionAddNew);
-    connect(actionAddNew, &QAction::triggered, this, &MainWindow::onAddNew);
-    setMenuBar(menuBar);
+    m_menuBar = std::make_unique<QMenuBar>(this);

```

Comment:
Using unique pointers can sometimes conflict with Qt's object ownership since the `MainWindow` will delete its children on deletion. Is there a particular reason to switch to these?

Reply:
Just to encourage people to care about memory management as sometimes Qt does not take ownership.

I tested this on macOS and I did not get any crash so we're probably fine. I can also revert this if you think it's safer.